pkg/ebpf/processor.go

```diff
@@ -0,0 +1,178 @@
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+// processLost handles lost events in a separate goroutine.
+func (t *Tracee) processLostEvents() {
+	logger.Debugw("Starting processLostEvents goroutine")
+	defer logger.Debugw("Stopped processLostEvents goroutine")
+
+	// Since this is an end-stage goroutine, it should be terminated when:
+	// - lostEvChannel is closed, or finally when;
+	// - internal done channel is closed (not ctx).
+	for {
+		select {
+		case lost, ok := <-t.lostEvChannel:
+			if !ok {
+				return // lostEvChannel is closed, lost is zero value
+			}
+
+			if err := t.stats.LostEvCount.Increment(lost); err != nil {
+				logger.Errorw("Incrementing lost event count", "error", err)
+			}
+			logger.Warnw(fmt.Sprintf("Lost %d events", lost))
+
+		// internal done channel is closed when Tracee is stopped via Tracee.Close()
+		case <-t.done:
+			return
+		}
+	}
+}
+
+// RegisterEventProcessor registers a new event processor for a specific event id.
+func (t *Tracee) RegisterEventProcessor(id events.ID, proc func(evt *trace.Event) error) {
+	if t.eventProcessor == nil {
+		t.eventProcessor = make(map[events.ID][]func(evt *trace.Event) error)
+	}
+	if t.eventProcessor[id] == nil {
+		t.eventProcessor[id] = make([]func(evt *trace.Event) error, 0)
+	}
+	t.eventProcessor[id] = append(t.eventProcessor[id], proc)
+}
+
+// registerEventProcessors registers all event processors, each to a specific event id.
+func (t *Tracee) registerEventProcessors() {
+	// Process events for the process tree before the regular event processing
+	switch t.config.ProcTree.Source {
+	case proctree.SourceBoth, proctree.SourceEvents:
+		t.RegisterEventProcessor(events.SchedProcessFork, t.procTreeForkProcessor)
+		t.RegisterEventProcessor(events.SchedProcessExec, t.procTreeExecProcessor)
+		t.RegisterEventProcessor(events.SchedProcessExit, t.procTreeExitProcessor)
+	}
+	// Process events from the regular pipeline
+	t.RegisterEventProcessor(events.VfsWrite, t.processWriteEvent)
+	t.RegisterEventProcessor(events.VfsWritev, t.processWriteEvent)
+	t.RegisterEventProcessor(events.KernelWrite, t.processWriteEvent)
+	t.RegisterEventProcessor(events.SecurityKernelReadFile, processKernelReadFile)
+	t.RegisterEventProcessor(events.SecurityPostReadFile, processKernelReadFile)
+	t.RegisterEventProcessor(events.SchedProcessExec, t.processSchedProcessExec)
+	t.RegisterEventProcessor(events.DoInitModule, t.processDoInitModule)
+	t.RegisterEventProcessor(events.HookedProcFops, t.processHookedProcFops)
+	t.RegisterEventProcessor(events.PrintNetSeqOps, t.processTriggeredEvent)
+	t.RegisterEventProcessor(events.PrintSyscallTable, t.processTriggeredEvent)
+	t.RegisterEventProcessor(events.PrintMemDump, t.processTriggeredEvent)
+	t.RegisterEventProcessor(events.PrintMemDump, t.processPrintMemDump)
+	// Run processors below for all events
+	t.RegisterEventProcessor(events.Undefined, t.normalizeTimings)
```

Comment:
To be more clear, maybe instead of Undefined we should call it "all"?

Reply:
I agree

Reply:
SGTM